File: api/model/dto/registrant.go
package dto

import "time"

type RegistrantGetByTicketID struct {
	Username   string    `json:"username"`
	AvatarURL  *string   `json:"picture"`
	IsVerified bool      `json:"is_verified"`
	CreatedAt  time.Time `json:"created_at"`
}

type PaginatedRegistrants struct {
	Data       []RegistrantGetByTicketID `json:"data"`
	Total      int64                     `json:"total"`
	Page       int                       `json:"page"`
	Limit      int                       `json:"limit"`
	TotalPages int                       `json:"total_pages"`
}

type RegistrantGetByUserID struct {
	ID         string    `json:"id"`
	TicketID   string    `json:"ticket_id"`
	URL        string    `json:"url"`
	Title      string    `json:"title"`
	ImageURL   *string   `json:"image"`
	IsVerified bool      `json:"is_verified"`
	CreatedAt  time.Time `json:"created_at"`
}

File: web/src/types/registrant.ts
export type RegistrantGetByTicketID = {
  username: string;
  picture: string;
  is_verified: boolean;
  created_at: string;
};

export type PaginatedRegistrants = {
  data: RegistrantGetByTicketID[];
  total: number;
  page: number;
  limit: number;
  total_pages: number;
};

export type RegistrantGetByUserID = {
  id: string;
  title: string;
  url: string;
  image: string;
  is_verified: string;
  created_at: string;
};
